pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/AbstractMultiPackagedProperty.java

```diff
@@ -4,48 +4,176 @@
 
 package net.sourceforge.pmd.lang.rule.properties;
 
+import static net.sourceforge.pmd.PropertyDescriptorField.LEGAL_PACKAGES;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import net.sourceforge.pmd.PropertyDescriptorField;
+
 /**
+ * Multi-valued property restricting the type of its values to some packages.
+ *
+ * @param <T> The type of the values
  *
  * @author Brian Remedios
- * @param <T>
+ * @version Refactored June 2017 (6.0.0)
  */
-public abstract class AbstractMultiPackagedProperty<T> extends AbstractPackagedProperty<T> {
+/* default */ abstract class AbstractMultiPackagedProperty<T> extends AbstractMultiValueProperty<T> {
 
+    /** Delimiter between values. */
     protected static final char DELIMITER = '|';
+    /** Required keys in the map. */
+    protected static final Map<PropertyDescriptorField, Boolean> PACKAGED_FIELD_TYPES_BY_KEY
+        = BasicPropertyDescriptorFactory.expectedFieldTypesWith(new PropertyDescriptorField[] {LEGAL_PACKAGES},
+                                                                new Boolean[] {false});
```

Comment:
I find the lockstep arrays as Map pattern unfortunate.  I'd much prefer a builder approach:
`expectedFieldTypes().with(LEGAL_PACKAGES, false).with(XXX, true).get()`

However, this is non-trivial to support.  Map literals would be even better. 

But now I'm not talking about Java. :)